src/pass/onnx_rewrite.cpp

```diff
@@ -17,6 +17,46 @@
 using namespace mlir;
 
 namespace {
+
+// Check whether an ArrayAttr contains non-zero values or not.
+bool hasNonZeroInArrayAttr(ArrayAttr attrs) {
+  bool allZeros = true;
+  if (attrs) {
+    for (auto attr: attrs.getValue()) {
+      if (attr.cast<IntegerAttr>().getInt() > 0) {
+        allZeros = false;
+        break;
+      }
+    }
+  }
+  return !allZeros;
+}
+
+// Create an ArrayAttr of IntergerAttr(s) of zero values.
+// This function is used for padding attribute in MaxPoolSingleOut.
+ArrayAttr createArrayAttrOfZeroWithTrail(
+    PatternRewriter &rewriter, ArrayAttr origAttrs, int trailCount) {
+  int nElements = origAttrs.getValue().size() + trailCount * 2;
+  SmallVector<int64_t, 4> vals(nElements, 0);
+  return rewriter.getI64ArrayAttr(vals);
+}
+
+// Pad a ArrayAttr with trails of zeros.
+// This function is used for padding attribute in MaxPoolSingleOut.
+ArrayAttr padArrayAttrWithZeroTrail(
```

Comment:
I would rename this to:

insertZerosForNonPaddedDims

And then I would insert a more extensive comment explaining how the zeros are inserted:

```
pads = [B1, B2, ... Bk, E1, E2, ..., Ek]

becomes:

pads = [0,... 0, B1, B2, ... Bk, 0,... 0, E1, E2, ..., Ek]
        |_____|                  |_____|
        nZeros                    nZeros
```